libibverbs/init.c

```diff
@@ -537,6 +537,7 @@ void ibverbs_device_put(struct ibv_device *dev)
 	struct verbs_device *verbs_device = verbs_get_device(dev);
 
 	if (atomic_fetch_sub(&verbs_device->refcount, 1) == 1) {
+		list_del(&verbs_device->entry);
```

Comment:
No.. the ownership of the list belongs to ibverbs_get_device_list, and it holds a reference so long as the device is in that list. When the device is dropped off the list the reference is dropped which permits the device to free. This would just be a double list_del in that path.

Reply:
@jgunthorpe Where do the "list_del" operation? When the refcount is 0, the entry should be freed.

Reply:
The refcount does not become 0 while the entry is on the list.

Reply:
atomic_fetch_sub will decrease 1 from verbs_device->refcount  and return it's original value. Since the original value is 1, so it current value should be 0, right?